bsip-0043.md

```diff
@@ -0,0 +1,96 @@
+Market fee reward is accumulated on the user account. The user decides when they want to claim the market fee reward and move it to their active balance. There is another operation created asset_claim_reward_operation. Each user pays network fee to call this operation.
+
+# Motivation
+
+To make promoting BitShares and bringing new users much more attractive to registrars and referrers by sharing UIAs and MPAs market fees with them.
+
+# Rationale
+When 'fill_order_operation' executing at the moment of market fee calculation there will be calculate the reward for the registrar used the parameter market_fee_reward_percent. Then this market_fee_reward_percent of the market fee is split between the registrar and referrer according to the referrer_rewards_percentage, which is set up during the new account registration by registrar (please see the parameters for register_account).
+
+# Specifications
+
+## market_fee_reward_percent asset option
+Percent of market fee that will be paid to buyer's registrar. Set by UIA issuer.
+
+## market_sharing_whitelist asset option
+An optional list of accounts (configurable by the UIA Issuer) could provide increased control over who is eligible to get market rewards.
+
+## New database and wallet api methods
+**get_account_reward(account, asset_symbol)** - Returns information about the reward amount for specific asset.
+**list_account_rewards(account)** - Returns information about the reward amount in various assets
+**claim_reward(account, asset_symbol, amount)** - Claim account's reward for the given asset.
+
+## account_reward_object
+A new BitShares object type in implementation space impl_account_reward_object_type = 2.18.X that tracks the rewards of a single account/asset pair
+```
+account_reward_object {
+owner_account,
+asset_type,
+reward_amount
+}
+```
+## account_reward_index
+A new index that stores objects of account_reward_object-type in graphene::database and allow random fast access to objects by given criteria
+```
+account_reward_index multi-index of account_reward_objects
+indexed_by
+[id]
+[owner_account, asset_type]
+[asset_type, reward_amount desc, owner_account]
+```
+## asset_claim_reward_operation
+A new operation used to transfer reward to the account's balance.
+```
+asset_claim_reward_operation {
+fee
+claiming_account
+amount_to_claim
+}
+```
+
+## graphene::chain::database new methods
+**get_reward(owner_account, asset_id)** - Retrieve a particular account's reward in a given asset
+**adjust_reward(account, delta)** - Adjust a particular account's reward in a given asset by a delta
+
+## graphene::chain::database modifications
+**pay_market_fees(seller, recv_asset_object, receives_amount)** - Split pay to asset issuer fee, registrar fee and referrer fee. If the registrar is not in the market_sharing_whitelist, split_pay will not happen and the entire fee goes to Asset issuer.
```

Comment:
Please specify behaviour if the new options are absent, or if the whitelist is empty.

Reply:
Also, note that fee is not paid out the the asset issuer but collected in the asset's fee pool (current behaviour).

Reply:
if market_fee_reward_percent = 0 or absent - the old mechanism is used. Market fees are accumulated in the asset and can be claimed.

If whitelist empty or absent - there is no filtering. That is like everyone is in the whitelist by default if it is empty.

Reply:
>Also, note that fee is not paid out the the asset issuer but collected in the asset's fee pool (current behaviour).

Yes, this mechanism stays the same way. The fees are accumulated, then claimed, not transferred. However, now not the whole market fee goes to the asset fee pool, but only a part of it: ```market_fee*(1 - market_fee_reward_percent)```. And ```market_fee* market_fee_reward_percent``` will get to the registar's fee pool .

Reply:
Thanks, I assumed as much. But you have to update the BSIP accordingly. Comments here are not part of the specification. :-)